tracr/craft/bases.py

```diff
@@ -156,27 +183,34 @@ def __post_init__(self):
   @property
   def num_dims(self) -> int:
     return len(self.basis)
+  
+  def get_basis_set(self):
+    if hasattr(self, 'basis_set'):
+      return self.basis_set
+    else:
+      self.basis_set = set(self.basis)
+      return self.basis_set
 
   def __contains__(self, item: Union[VectorInBasis, BasisDirection]) -> bool:
     if isinstance(item, BasisDirection):
-      return item in self.basis
-
-    return set(self.basis) == set(item.basis_directions)
+      return item in self.get_basis_set()
+    # item is a VectorInBasis, so it must also be sorted under the same ordering so we can just check the lists
```

Comment:
yes list is quicker, since the basis directions are ordered anyway, otherwise you need to do an exhaustive search over the set rather than being able to exit as soon as you find a missmatch